### DISCUSSION TITLE
pool not found

### DISCUSSION THREAD

**Discussion by ravenclaw78** [Sithuk/ubuntu-server-zfsbootmenu#44] [General]
After successfully install and reboot refind started, zfsbm booted, but could not find the pool to boot.
I managed to import it in the recovery console with
```
zpool import -d /dev/sda3 rpool
```
After exiting the recovery console, it could boot the kernel, but
then the same thing happened in initramfs.

With 
```
/sbin/zpool import -d /dev/sda3 rpool
exit
```
it finally boots correctly.

Any idea on how to fix this?

**Comment by Sithuk** [comment]
I haven't had that error before. Do you have to do that every boot?

**Reply by ravenclaw78** [reply]
Every time!

I just remembered, that when I first started the script, it ended prematurely with an error that the changed partition layout is not reflected in ram. I then just restarted, run the script with the exact same parameters and it run to completion.

Could that be the cause of the problem?

When I boot it looks like the [screenshot](https://discourse.practicalzfs.com/uploads/default/original/1X/fd8ece503e8ab2b54cd9c041582cac93ed81357a.png) in this [thread](https://discourse.practicalzfs.com/t/zfsbootmenu-unable-to-import-pool/1424)

**Reply by Sithuk** [reply]
The script is also currently building zfsbootmenu from the github master. The reason is to take advantage of a recent bug fix to allow  remote access to a luks encrypted pool. The zfsbootmenu master could be introducing issues. If you don't need the remote unlock for a luks encrypted drive, you could try a re-install using the previous zfsbootmenu release version? You can adapt the script to install it by changing lines 957 and 958 to the following.

#download_zbm_git_master
download_zbm_release


If you have the installation log that would be helpful too. The log should be saved at "/var/log". Be sure to remove any sensitive information such as passwords from the log if you post it.

**Reply by ravenclaw78** [reply]
Reinstalling the laptop is currently not possible (I already configured it fully), but I will try in an VM.
I bootet from an usb stick with zbm release on it, it also could not boot the pool without importing it first.

I found the following error, but as I don't use luks encryption this may be irrelevant.
```
## dracut-install: ERROR: installing 'cryptsetup'
## dracut[E]: FAILED: /usr/lib/dracut/dracut-install -D /var/tmp/dracut.4Z054U/initramfs -a cryptsetup rmdir readlink umount
```

Can you explane to me what the hostid has to do with zbm? Is there something I can check?
```
zfs set 'org.zfsbootmenu:commandline=spl_hostid=368ffa89 ro quiet' rpool/ROOT
```

[ubuntu_setup_zfs_root.log](https://github.com/user-attachments/files/15828385/ubuntu_setup_zfs_root.log)

**Comment by ravenclaw78** [comment]
I did a few tests in a kvm, and the first one ended like my laptop install.
```
Clearing partition table on disk virtio-56546643.
Creating new GPT entries in memory.
GPT data structures destroyed! You may now partition the disk using fdisk or
other utilities.
Warning: Unable to open /dev/sr0 read-write (Read-only file system).  /dev/sr0 has been opened read-only.
Creating partitions on disk virtio-56546643.
Creating new GPT entries in memory.
The operation has completed successfully.
The operation has completed successfully.
The operation has completed successfully.
Warning: Unable to open /dev/sr0 read-write (Read-only file system).  /dev/sr0 has been opened read-only.
cannot resolve path '/dev/disk/by-id/virtio-56546643-part3'
```
Manual verification showed the partitions are correctly set up and 'there'.
```
Disk /dev/vda: 25 GiB, 26843545600 bytes, 52428800 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 30F24B19-EF02-4911-8DF6-56C56706F1CB

Device       Start      End  Sectors  Size Type
/dev/vda1     2048  1050623  1048576  512M EFI System
/dev/vda2  1050624  2361343  1310720  640M Linux swap
/dev/vda3  2361344 52428766 50067423 23.9G Solaris root

root@ubuntu:/home/ubuntu# ls -l /dev/disk/by-id/virtio-56546643-part3
lrwxrwxrwx 1 root root 10 Jun 27 13:46 /dev/disk/by-id/virtio-56546643-part3 -> ../../vda3
root@ubuntu:/home/ubuntu# ls -l /dev/vda3
brw-rw---- 1 root disk 253, 3 Jun 27 13:46 /dev/vda3
```

After removing the comment from all ```sleep 2``` commands (each one after updating partition table), in the script, the install finished.

I will install another bare metal these days and report back if that solved it.

**Reply by Sithuk** [reply]
Thank you for the feedback. I commented out the sleep commands in the hope that running partprobe meant that the artifical delay wasn't needed. It looks like the sleep commands are needed. I will consider re-implementing them in the script.
